packages/mui-system/src/Stack/createStack.tsx

```diff
@@ -51,7 +51,12 @@ function useThemePropsDefault<T extends {}>(props: T) {
  * [1,0,2,0,3]
  */
 function joinChildren(children: React.ReactNode, separator: React.ReactElement<unknown>) {
-  const childrenArray = React.Children.toArray(children).filter(Boolean);
+  const childrenArray = React.Children.toArray(children).filter((child) => {
+    if (child === 0) {
+      return true;
+    }
+    return !!child;
```

Comment:
Why is this not using `getValidReactChildren`?

Reply:
>Shouldn't this be using getValidReactChildren()?

Hmm not sure about this, `React.isValidElement` is being used inside `getValidReactChildren` and  `React.isValidElement`  is returning false when any number or strings are passed (attached reference for this below). So if we want to directly render numbers or strings without wrapping in html tags/React components, i think these would get skipped.



<img width="955" alt="image" src="https://github.com/user-attachments/assets/b23cc9f9-99bb-47b4-86fa-c95ca1dc4ce7">

Reply:
And then, we could use https://github.com/mui/material-ui/blob/ca637077e091b7f1f6e591ded5a453edcc784204/packages/mui-material/src/Tooltip/Tooltip.js#L562 as inspiration on how to detect this.